bandit.js

```diff
@@ -0,0 +1,141 @@
+require('dotenv').config({silent: true})
+
+var Connection = require('./connection')
+
+
+class Bandit {
+  constructor() {
+    // NOTE: Be very careful here. If all references to SLACK_ID are not accounted for
+    // it will result in an infinite loop.
+    this.SLACK_ID = "<@U017GNL24JY>"
+    this.SLACK_CHANNEL = process.env.SLACK_CHANNEL
+    this.TOKEN = process.env.TOKEN
+
+    this.connection = new Connection({ token: this.TOKEN })
+    this.connection.listen({
+      channel: this.SLACK_CHANNEL,
+      callback: this.handleMessage.bind(this),
+    })
+
+    this.commands = ["start", "guess"];
+    this.state = {};
+  }
+
+  send(message) {
+    this.connection.send({ channel: this.SLACK_CHANNEL, message })
+  }
+
+  /**
+   * 
+   * @param {String} text 
+   * @param {Array<String>} words 
+   * @returns {Boolean}
+   */
+  containsAny(text, words) {
+    for (let word of words) {
+      if (text.includes(word)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  sendNoCommandsMessage(message)
+  {
+    this.send(`<@${message.user}> Unrecognized command. Available commands are: ${this.commands.join(", ")}`);
+  }
+
+  sendStartMessage(message) {
+    let options = Object.keys(this.state[message.user].rewards).join(", ");
+    this.send(`<@${message.user}> You've started a game.
+      \ Your options are ${options}. 
+      \ Attempt a guess by writing \`${this.SLACK_ID} guess {option}\`\n
+      \ begin ${options}`);
+  }
+
+  sendGuessMessage(message) {
+    let options = Object.keys(this.state[message.user].rewards);
+    let playerState = this.state[message.user];
+    let guess = message.text.replace(this.SLACK_ID, "");
+    guess = guess.replace("guess", "").trim();
+
+    if (this.containsAny(guess, options)) {
+      let reward = playerState.rewards[guess];
+      playerState.total += reward;
+
+      this.send(`<@${message.user}> You guessed "${guess}".
+        \ Reward: ${reward}. 
+        \ Total Score: ${playerState.total}. 
+        \ Moves Remianing: ${--playerState.moves}\n
+        \ reward ${reward}, ${guess}, ${playerState.total}, ${playerState.moves}`)
+
+      if (playerState.moves <= 0) {
+        delete this.state[message.user];
+        this.send(`<@${message.user}> Game over.
+          \ Your score was: ${playerState.total}`);
+      }
+    }
+
+    else {
+      this.send(`<@${message.user}> You didn't take a possible action.
+        \ Your options are ${options.join(", ")}. 
+        \ Attempt a guess by writing \`${this.SLACK_ID} guess {option}\``);
+    }
+  }
+
+  generateRewards() {
+    let words = ["A", "B", "C", "D", "E"];
+    let rewards = {};
+
+    for (let word of words) {
+      // Random number between -5..5
+      rewards[word] = Math.random() * 10 - 5;
```

Comment:
From examples of the bandit problem I've come across, it seems like there's no notion of a negative reward after pulling a lever. I think what we want to store here is a probability that guessing a given word would return a "reward". So the value of each successful guess is the same, but what we need to figure out through the game is which word will return a success most of the time.

Reply:
> So the value of each successful guess is the same, but what we need to figure out through the game is which word will return a success most of the time

I think we discussed that each move would return the same result every time it was guessed in the first iteration. Are you suggesting the values would change each time a move is guessed?

Reply:
> I think we discussed that each move would return the same result every time it was guessed in the first iteration. 

👍 I missed this discussion earlier, but @HeyZoos caught me up.

> Are you suggesting the values would change each time a move is guessed?

Nope -- the values would stay the same, say +1 point per successful guess, but the probability of guessing a word and getting that point would be different for each word. This graphic probably explains it better than I did:

![image](https://www.topbots.com/wp-content/uploads/2019/05/cover_multi_armed_bandit_1600px_web-1280x640.jpg)

But it sounds like with this first iteration, we're going to stick with a set amount of points for each choice (per round) rather than a probability.